### PR TITLE
Stickin it to Mason

### PR DESCRIPTION
**Pull request by jimtheplant**
🖕

### CODE REVIEW

File: xdx.py [hunk 1]
```diff
@@ -5,16 +5,17 @@
...
-    else:
-        return roll_baby_roll(left-1, size) + randy(1, size)
+	for i in range(left):
+		yield randy(1,size)
+
 
 def main():
     dice = sys.argv[1].split('d')
-    print(roll_baby_roll(int(dice[0]), int(dice[1])))
+    tot = 0
```
Comment (masonseeger) [inline xdx.py]: can this variable name be clearer?

File: xdx.py [hunk 2]
```diff
@@ -5,16 +5,17 @@
...
 def roll_baby_roll(left, size):
-    if left < 1:
-        return 0
-    elif left == 1:
-        return randy(1, size)
-    else:
-        return roll_baby_roll(left-1, size) + randy(1, size)
+	for i in range(left):
+		yield randy(1,size)
+
```
Comment (masonseeger) [inline xdx.py]: nit: extra line

**Comment by rachel-camp** [comment]
+1

**Review by rachel-camp — DISMISSED** [review]

File: xdx.py [hunk 3]
```diff
@@ -5,16 +5,17 @@
...
 
 def main():
     dice = sys.argv[1].split('d')
-    print(roll_baby_roll(int(dice[0]), int(dice[1])))
+    tot = 0
+    for dice in roll_baby_roll(int(dice[0]), int(dice[1])):
+        print(dice)
```
Comment (masonseeger) [inline xdx.py]: Can you add a roll num here so there aren't too many numbers being returned one after the other?

Comment (jimtheplant) [reply xdx.py]: No. Git good 😎

Comment (masonseeger) [reply xdx.py]: EX: Roll 1: 
Roll 2:
Roll 3:
...

Comment (jimtheplant) [reply xdx.py]: [Happy](https://github.com/masonseeger/xdx-superiority-project/pull/1/commits/181890ddbf7c893c1c2e368affe72c78887eea76)?

File: xdx.py [hunk 4]
```diff
@@ -5,16 +5,16 @@
...
-    else:
-        return roll_baby_roll(left-1, size) + randy(1, size)
+    for i in range(left):
+      yield i, randy(1,size)
 
 def main():
     dice = sys.argv[1].split('d')
-    print(roll_baby_roll(int(dice[0]), int(dice[1])))
+    total = 0
+    for dice in roll_baby_roll(int(dice[0]), int(dice[1])):
```
Comment (masonseeger) [inline xdx.py]: It's a bit confusing having dice as vars in both places here

Comment (jimtheplant) [reply xdx.py]: https://github.com/masonseeger/xdx-superiority-project/pull/1/commits/da9b75ec2b023a509decc77e27fad60667fac6ad

**Review by masonseeger — APPROVED** [review]

**Pull request merged**